### PR TITLE
Updated setup.py to check argv

### PR DESCRIPTION
**Pull request by souravsingh**
Updated setup.py to check for value in arguments and detect if installation is required.

### CODE REVIEW

**Comment by souravsingh** [comment]
I am not able to figure out the cause for build test failures.

**Comment by anderspitman** [comment]
Hi @souravsingh, looks like the build is failing because you're using `sys.argv` without importing `sys` first. Adding `import sys` to the top should fix it.

**Comment by souravsingh** [comment]
My bad,I didn't figure it out.Thanks for pointing it out @anderspitman

**Comment by souravsingh** [comment]
I am getting PEP8 errors due to which tests are failing.

**Comment by anderspitman** [comment]
This time it says one of the lines you added is too long. You'll need to split it.

Probably the best thing for you to do is set up a development environment for testing skbio locally:
https://github.com/biocore/scikit-bio/blob/master/CONTRIBUTING.md

We don't yet have a way to run everything exactly the way Travis CI does, but that will get you a lot of the way there.

File: setup.py [hunk 1]
```diff
@@ -12,15 +12,29 @@
...
+else:
+    setup_requires = ['numpy >= 1.9.2']
 
 # Bootstrap setup.py with numpy
 # Huge thanks to coldfix's solution
 # http://stackoverflow.com/a/21621689/579416
-class build_ext(_build_ext):
+ 
+ 
+ class build_ext(_build_ext):
```
Comment (jairideout) [inline setup.py]: Can you revert the changes to these lines? It is causing the Travis build to error:

```
Processing /home/travis/build/biocore/scikit-bio
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 20, in <module>
      File "/tmp/pip-7RadPw-build/setup.py", line 37
        class build_ext(_build_ext):
                                   ^
    IndentationError: unindent does not match any outer indentation level
```

File: setup.py [hunk 2]
```diff
@@ -12,15 +12,29 @@
...
+if any(
+    arg.startswith('--') and arg not in [
+        '--verbose',
+        '--quiet',
+        '--dry-run',
+        '--no-user-cfg',
+        '--command-packages'] for arg in sys.argv):
+    setup_requires = []
+else:
+    setup_requires = ['numpy >= 1.9.2']
```
Comment (jairideout) [inline setup.py]: You'll need to modify the `setup()` call below to use this `setup_requires` variable.

**Comment by wasade** [comment]
Just a note to tie the issue, I believe this fixes #997

**Comment by souravsingh** [comment]
@jairideout Correct me if I am wrong We have to add this code to the ````setup()```` right?
````if len(sys.argv) == 1 or (
            len(sys.argv) >= 2 and ('--help' in sys.argv[1:] or
                                    sys.argv[1] in ('--help-commands',
                                                    'egg_info',
                                                    '--version',
                                                    'clean'))) ````

**Comment by jairideout** [comment]
@souravsingh we are closing this pull request and associated issue in favor of #1296. Thanks for your work on this!
